addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CrossSiteScriptingScanRule.java

```diff
@@ -588,16 +588,19 @@ public void scan(HttpMessage msg, String param, String value) {
                             if (contexts2 == null) {
                                 return;
                             }
-                            if (contexts2.size() > 0) {
-                                // Yep, its vulnerable
-                                newAlert()
-                                        .setConfidence(Alert.CONFIDENCE_MEDIUM)
-                                        .setParam(param)
-                                        .setAttack(contexts2.get(0).getTarget())
-                                        .setEvidence(contexts2.get(0).getTarget())
-                                        .setMessage(contexts2.get(0).getMsg())
-                                        .raise();
-                                attackWorked = true;
+                            for (HtmlContext ctx : contexts2) {
+                                if (ctx.getSurroundingQuote().isEmpty()) {
+                                    // Yep, its vulnerable
+                                    newAlert()
+                                            .setConfidence(Alert.CONFIDENCE_MEDIUM)
+                                            .setParam(param)
+                                            .setAttack(contexts2.get(0).getTarget())
+                                            .setEvidence(contexts2.get(0).getTarget())
+                                            .setMessage(contexts2.get(0).getMsg())
```

Comment:
Since this is looping through a list shouldn’t `ctx` be used instead of the zero’th element?

Reply:
Good catch.

Reply:
The target is the same for all contexts (since the attack is the same). Although for correctness better use `ctx`, will push that change.